create docker build action

guidoffm opened the pull request:
Create a GitHub action that does a multi platform Docker build and publishes the image to GitHub

guidoffm committed:
Add docker-build.yml

guidoffm committed:
Update branch name in docker/build.yml

r00t- commented:
thanks,
can you say how this compares to https://github.com/volkszaehler/vzlogger/pull/478 ?
(that one has been stalled, not sure why right now.)

guidoffm commented:
> thanks, can you say how this compares to #478 ? (that one has been stalled, not sure why right now.)

I have commented in https://github.com/volkszaehler/vzlogger/pull/478

StefanSchoof commented on .github/workflows/docker-build.yml:
```diff
@@ -0,0 +1,102 @@
+
+env:
+  # Use docker.io for Docker Hub if empty
+  REGISTRY: ghcr.io
```
down below the password is set to `secrets.GITHUB_TOKEN` so here no real change possible. I think the project should choose on registry (ghcr.io is fine) and remove this from env.

StefanSchoof commented on .github/workflows/docker-build.yml:
```diff
@@ -0,0 +1,102 @@
+      # multi-platform images and export cache
+      # https://github.com/docker/setup-buildx-action
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@f95db51fddba0c2d1ec667646a06c2ce06100226 # v3.0.0
```
Using the sha is not done for the other github actions. Having here sha makes this without a dependabot or renovate hard to maintain. Do not know what the best solution is.

StefanSchoof commented on .github/workflows/docker-build.yml:
```diff
@@ -0,0 +1,102 @@
+        id: build-and-push
+        uses: docker/build-push-action@0565240e2d4ab88bba5387d719585280857ece09 # v5.0.0
+        with:
+          context: .
```
Why use checkout and context instead of the default git access of buildx?

guidoffm committed:
Merge branch 'master' into feature/create-docker-build-action

r00t- commented:
guessing from from the boilerplate comments, your workflow definition seems to be created based on some example, can you link that?
(or even commit the original and then your changes on top?)

r00t- commented:
ping @guidoffm

guidoffm commented:
What you can do:

Go to the "Actions tab". Then hit the button "New Workflow". Then choose the template "Publish Docker Container" and hit the "Configure" button.

oeiber commented:
How can I help to get build action implemented?

r00t- commented:
@oeiber:
you could address any of the comments above, either by commenting with suggestions here, or creating an MR of your own.

personally i'm no expert on this and it's not a priority for me.
my comment above was mostly that i don't like how i cannot tell which of this code is autogenerated and what was manually adjusted, so i suggested to put those into separate commits.
Stefan as author of the previous older MR also had some good questions that remained unanswered.

guidoffm committed:
docker-build: add QEMU setup and armv7 platform

Register QEMU emulators via docker/setup-qemu-action so buildx can
cross-build non-host platforms on the amd64 runner, and extend the
build matrix to include linux/arm/v7 alongside amd64 and arm64 to
cover 32-bit Raspberry Pi 2B/3 deployments.

guidoffm committed:
docker-build: allow manual trigger via workflow_dispatch

guidoffm committed:
docker-build: bump actions to Node-24-compatible versions

GitHub forces Node.js 24 on action runners from 2026-06-16 and
removes Node.js 20 entirely on 2026-09-16. Bump:
  actions/checkout              v4    -> v6
  sigstore/cosign-installer     v3.5  -> v4
  docker/setup-qemu-action      v3    -> v4
  docker/setup-buildx-action    v3.0  -> v4
  docker/login-action           v3.0  -> v4
  docker/metadata-action        v5.0  -> v6
  docker/build-push-action      v5.0  -> v7

Drop the previous SHA pins in favour of clean major-version tags
for readability; all bumped actions are maintained by docker /
sigstore / github orgs.

guidoffm committed:
docker-build: pin cosign-installer to v4.1.2

sigstore/cosign-installer does not maintain a floating major tag,
so v4 doesn't resolve. Use the explicit latest patch version.
